app/V1Module/presenters/AssignmentsPresenter.php

```diff
@@ -698,7 +704,7 @@ public function actionDownloadBestSolutionsArchive(string $id)
 
             $studentDir = Strings::webalize("{$student->getLastName()}_{$student->getFirstName()}_{$student->getId()}");
             foreach ($solution->getSolution()->getFiles() as $file) {
-                $files[$file->getLocalFilePath()] = $studentDir . "/" . $file->getName();
+                $files[$studentDir . "/" . $file->getName()] = $file->getFile($this->fileStorage);
```

Comment:
> $file->getFile($this->fileStorage

this seems like a bit reversed logic

Reply:
Depends on point of view, I suppose. However, the file returned from storage is wrapped in ImmutableFile object (abstraction over files), so we have no choice. Furthermore, I see logic in having the presentation name as key and the actual file (object in this case) as value.